node/resolvers/profile/index.ts

```diff
@@ -23,6 +23,16 @@ export const mutations = {
 
   uploadProfilePicture: (_, { file }, context) =>
     updateProfilePicture(context, file),
+
+  subscribeNewsletter: async (_, { email }, context: Context) => {
+    const profile = context.dataSources.profile
+
+    await profile.updatePersonalPreferences(email, {
+      isNewsletterOptIn: "True",
+    })
+
+    return true
```

Comment:
Is returning true really necessary? Also, since it's a newsletter you might eventually have to create an unsubscribe mutation for it, wouldn't it be better to receive the "isNewsletterOptIn" as an argument and have just one updateNewsLetterPreferences mutation? Also, since you updating profile info, you might want to use the currentProfile that it is passed on the context by the withCurrentProfile directive instead of receiving the user email as an arg.

Reply:
@salesfelipe 
The idea is to have an input to type an email to start receiving newsletters. The user shouldn't have to login to do that. 

`withCurrentProfile` takes that into account and that is not the desirable behaviour.

But to unsubscribe you might need to login to do that.

About returning `true`, I don't know.. should I return `undefined`?

Reply:
Not sure either, I'm just worried about the consequences of inserting info com the CL table without passing through the profileSystem, because you won't have the userId and when the user actually tries to buy something, the vtexID will try to create a user  with that email, but that email will be already registered on the CL table? Dunno, if they check first and then update if already exists, you might want to check on that first.

Reply:
Another thing, since it's a public api, if I have your email, I would be able to subscribe you on my newsletter without any authorization.